components/Buttons/src/MDCButton.m

```diff
@@ -676,7 +676,9 @@ - (CGFloat)elevationForState:(UIControlState)state {
 
 - (void)setElevation:(CGFloat)elevation forState:(UIControlState)state {
   _userElevations[@(state)] = @(elevation);
-  self.layer.elevation = [self elevationForState:self.state];
+  if (self.state == state) {
```

Comment:
I think you mean !=

but be careful because we want the default elevation to also be set.

Reply:
No, we only want to update elevation-related properties if the elevation for the current state has changed. Otherwise those properties will be updated on the next state change.